source/Halibut/Transport/Proxy/HttpProxyClient.cs

```diff
@@ -344,6 +425,18 @@ void WaitForData(NetworkStream stream)
                     throw new ProxyException(string.Format("A timeout while waiting for the proxy server at {0} on port {1} to respond.", Utils.GetHost(TcpClient), Utils.GetPort(TcpClient)), true);
             }
         }
+        
+        async Task WaitForDataAsync(NetworkStream stream, CancellationToken cancellationToken)
```

Comment:
Could this become redundant with the `TimeoutStream`?

Reply:
Yes I think so. I wont change this however since we are not changing behaviour.